### PR TITLE
feat: allow setting the window title in hover.yaml

### LINKED ISSUE #433: how to custom projectName and icon
how to custom projectName and icon

### PR DESCRIPTION
**Pull request by pchampio**
needs non-breaking and backward compatible hover PR https://github.com/go-flutter-desktop/hover/pull/138
fixes: #433

### CODE REVIEW

**Comment by GeertJohan** [comment]
What if the user want's to set the title from the app? e.g. based on location within the app.. e.g.:

"My Awesome App - Profile Settings"

I think it could be nice if we can leverage the hover application name to set the window title during startup, but I think it should still be overwritten by whatever is told us from the dart side. That would be easily achieved by undoing the removes in platform.go in this PR.

File: platform.go [hunk 1]
```diff
@@ -97,18 +97,6 @@ func (p *platformPlugin) handleClipboardGetData(arguments interface{}) (reply in
...
-	appSwitcherDescription := struct {
-		Label        string `json:"label"`
-		PrimaryColor int64  `json:"primaryColor"`
-	}{}
-	err = json.Unmarshal(arguments.(json.RawMessage), &appSwitcherDescription)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to decode arguments")
-	}
-	p.glfwTasker.Do(func() {
-		p.window.SetTitle(appSwitcherDescription.Label)
```
Comment (GeertJohan) [inline platform.go:109]: I think we should still do this.

Comment (pchampio) [reply platform.go:109]: By default, `appSwitcherDescription.Label` always returns the `name` field of pubspec.yaml.

Comment (pchampio) [reply platform.go:109]: This field is for mobile devices, not suited for desktop: https://github.com/flutter/flutter/blob/f7a6a7906be96d2288f5d63a5a54c515a6e987fe/packages/flutter/lib/src/services/system_channels.dart#L79-L88

Comment (pchampio) [reply platform.go:109]: The best we could have is to use: https://github.com/flutter/flutter/blob/7168421e9f8f4122bfe1036e2f398c6392dd9d17/packages/flutter/lib/src/widgets/route_notification_messages.dart#L30-L38
But this is currently only possible on web.
I'll open a PR.

Comment (GeertJohan) [reply platform.go:109]: Alright! I never got that this was not the correct handling of the switcher description. Let's indeed handle it correctly and wait for the window-specific title setter method.

**Comment by jinfagang** [comment]
Currently how to change the title name?

**Comment by GeertJohan** [comment]
@pchampio This is waiting on a feature from flutter right?

**Comment by pchampio** [comment]
@GeertJohan Nope, this is fixed in the latest version of hover.
You can tweak the window title (and other) by changing those line from the go/hover.yaml file https://github.com/go-flutter-desktop/hover/blob/1d24f84e8b2110073d57506022730b630e7ea152/assets/app/hover.yaml.tmpl#L1-L3

**Comment by pchampio** [comment]
You can also craft your own flutter/navigation plugin that listen for flutter navigation changes and update the glfw window title (through GLFW plugin)
https://github.com/go-flutter-desktop/go-flutter/wiki

**Comment by GeertJohan** [comment]
I think I meant https://github.com/flutter/flutter/issues/30712

**Comment by pchampio** [comment]
Ohh, yea, that another feature!
I have subscribed to a bunch of desktop related issues on the flutter issue tracker...
Once they get close, I usually implement them or open a issue on go-flutter. (Just like I did with the text input)

**Pull request merged**